bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/StringListType.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.smarthome.core.library.types;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Formatter;
+import java.util.List;
+
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+
+/**
+ * This type can be used for items that are dealing with telephony functionality.
+ *
+ * @author Thomas.Eichstaedt-Engelen / Initially CallType
+ * @author Gaël L'hopital - port to Eclipse SmartHome
+ *
+ */
+public class StringListType implements Command, State {
+
+    protected List<String> typeDetails;
+
+    // constants
+    static final public String DELIMITER = ",";
+    static final public String ESCAPED_DELIMITER = "\\" + DELIMITER;
+    static final public String REGEX_SPLITTER = "(?<!\\\\)" + DELIMITER;
+
+    public StringListType(StringType... rows) {
+        typeDetails = new ArrayList<String>();
```

Comment:
We can use `new ArrayList<String>(rows.length)` to prevent memory reallocation and copy.